src/JustSaying/JustSayingBus.cs

```diff
@@ -247,6 +247,12 @@ private IMessagePublisher GetPublisherForMessage(Message message)
             CancellationToken cancellationToken)
         {
             attemptCount++;
+
+            if (message.TimeStampWithOffset == null)
```

Comment:
I wonder if we should move the set of the existing TimeStamp property here as well, so that both are populated at the same time. Given the models package is being versioned as well, we can make that change safely.